### PR TITLE
allow for multiple lcds that get rotated in case of an error

### PR DESCRIPTION
**Pull request by dee-effes**

### CODE REVIEW

File: feeder/src/vote.ts [hunk 1]
```diff
@@ -295,33 +295,69 @@ interface VoteArgs {
...
+      client = new LCDClient({
+        URL: args.lcdAddress[currentLCDIndex],
+        chainID: args.chainID,
+        gasPrices: args.gasPrices,
+      })
+      wallet = new Wallet(client, rawKey)
+      console.log(
+        'Switched to LCD address ' + currentLCDIndex + '(' + args.lcdAddress[currentLCDIndex] + ')'
+      )
+    }
```
Comment (freshfab) [inline feeder/src/vote.ts]: I'd suggest putting the LCD switching logic into its own separate function `switchLCD`, which then returns a new LCDClient. In that case, you'd just need to put in the function call in the catch clause below. Makes things more readable. Other than that, looks good to me 👍🏼

File: feeder/src/vote.ts [hunk 2]
```diff
@@ -295,33 +295,69 @@ interface VoteArgs {
...
-        console.error(err.message, err.response.data)
-      } else {
-        console.error(err.message)
+    if (switchLCD) {
+      switchLCD = false
+      if (++currentLCDIndex > maxLCDIndex) {
+        currentLCDIndex = 0
       }
+      client = new LCDClient({
+        URL: args.lcdAddress[currentLCDIndex],
```
Comment (freshfab) [inline feeder/src/vote.ts]: Shouldn't `args.lcdAddress` be of type `string[]` instead of `string`?

Comment (dee-effes) [reply feeder/src/vote.ts]: It' actually a string. I changed the type in VoteArgs.

File: feeder/src/vote.ts (hunk 1 above)
Comment (dee-effes) [reply feeder/src/vote.ts]: I moved the lcd switch logic to its own function

File: feeder/src/vote.ts [hunk 3]
```diff
@@ -295,33 +295,88 @@ interface VoteArgs {
...
+    URL: args.lcdAddress[currentLCDIndex],
     chainID: args.chainID,
     gasPrices: args.gasPrices,
   })
   const rawKey: RawKey = await initKey(args.keyPath, args.password)
   const valAddrs = args.validator || [rawKey.valAddress]
   const voterAddr = rawKey.accAddress
-  const wallet = new Wallet(client, rawKey)
+  let wallet = new Wallet(client, rawKey)
+  let switchLCD = false
```
Comment (freshfab) [inline feeder/src/vote.ts]: AFAICS, the switchLCD variable is not needed. Just call `rotateLCD` in the catch clause below.

**Pull request merged**